halo2_proofs/src/plonk/circuit.rs

```diff
@@ -1103,7 +1106,7 @@ impl<F: Field> Expression<F> {
     }
 }
 
-impl<F: std::fmt::Debug> std::fmt::Debug for Expression<F> {
+impl<F: ff::Field> std::fmt::Debug for Expression<F> {
```

Comment:
[critical-3]
their are some gagetes in zkevm-circuit e.g. [is_equal](https://github.com/scroll-tech/zkevm-circuits/blob/1093f5ecd2fee40fb7edb59df44bb30937a4e771/gadgets/src/is_equal.rs#L27-L34) has std::fmt::Debug trait, this change triggers compilation errors like:
```
28 | #[derive(Clone, Debug)]
| ----- in this derive macro expansion
...
33 | pub is_equal_expression: Expression<F>,
| ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `halo2_proofs::arithmetic::Field` is not implemented for `F`
|
= note: required for `Expression<F>` to implement `Debug`
```